docs/overview/architecture/proposal-functions.md

```diff
@@ -0,0 +1,37 @@
+The [Proposal.sol](../../../src/proposals/Proposal.sol) file contains a set of functions that all governance models inherit. There are three governance models: Bravo, Multisig and Timelock already implemented in the example repository. To create a Proposal with FPS, you can inherit one of the governance types, such as [MultisigProposal.sol](../../../src/proposals/MultisigProposal.sol) and override the needed functions. Alternatively, you can create a new governance model in your repository that inherits the `Proposal` contract and then a proposal can be created using a newly formed governance model not already inside the repo. The following is a list of functions all proposals must implement:
+
+-   `function name() public` Override this function to define the proposal name. This function should be overridden at the final proposal level.
+
+-   `function description() public` Override this function to define the proposal description. This function should be overridden at the final proposal level.
+
+-   `function deploy() public`: Defines new contract deployments. Newly deployed contracts must be added to the `Addresses` contract instance through the setters methods. This function should be overridden only when there are deployments to be made in the proposal.
+
+-   `function afterDeployMock() public`: Specifies post-deployment actions. Such actions can include wiring contracts together, transferring ownership rights, or invoking setter functions as the deployer.
+
+-   `function build() public`: Creates the proposal actions and saves them to storage in the proposal contract.
+
+-   `function simulate() public`: Executes the actions that were previously saved during the `build` step. This function's execution is dependent on the successful execution of the `build` function. Without calling `build` first, the `_simulateActions()` function becomes ineffectual as there would be no predefined actions to execute.
+
+-   `function run() public`: Simulates proposal execution against a forked mainnet or locally and can be used in integration tests or scripts. This function can be overridden at the final proposal level where it can be used to set the environment, for eg: addresses object, primary fork id. Make sure to call the `run()` from the Proposal contract as it performs key actions.
```

Comment:
```suggestion
-   `function run() public`: Simulates the execution of a proposal against a forked mainnet or locally, and it can be used in integration tests or scripts. It must be overridden at the final proposal level to set up the environment, including creating the addresses instance, setting the governance contract, and creating the primary fork. When overriding, make sure to call `super.run()`.
```